Sources/Identity/ProductPaidPrice.swift

```diff
@@ -22,12 +22,46 @@ import Foundation
     /// Amount paid
     @objc public let amount: Double
 
+    /// Formatted price of the item, including its currency sign. For example $3.00.
+    @objc public let formatted: String
+
+    private static let numberFormatter: NumberFormatter = {
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .currency
+        return formatter
+    }()
+
     /// ProductPaidPrice initialiser
     /// - Parameters:
     ///   - currency: Currency paid
     ///   - amount: Amount paid
-    public init(currency: String, amount: Double) {
+    ///   - formatted: Formatted price string with currency symbol
+    public init(currency: String, amount: Double, formatted: String) {
         self.currency = currency
         self.amount = amount
+        self.formatted = formatted
+    }
+
+    /// Convenience initializer that formats the price using the provided locale
+    /// - Parameters:
+    ///   - currency: Currency code (e.g., "USD", "EUR")
+    ///   - amount: Amount as a decimal value
+    ///   - locale: Locale for formatting (defaults to current locale)
+    public convenience init(currency: String, amount: Double, locale: Locale = .current) {
+        let formatted = Self.formatPrice(amount: amount, currency: currency, locale: locale)
+        self.init(currency: currency, amount: amount, formatted: formatted)
+    }
+
+    /// Formats a price with currency using NumberFormatter
+    /// - Parameters:
+    ///   - amount: The price amount
+    ///   - currency: The currency code
+    ///   - locale: The locale for formatting
+    /// - Returns: Formatted price string (e.g., "$3.00", "€7.99")
+    static func formatPrice(amount: Double, currency: String, locale: Locale = .current) -> String {
+        numberFormatter.locale = locale
+        numberFormatter.currencyCode = currency
```

Comment:
Nitpick and FFTI, but I've been thinking that this could in theory cause issues if we create ProductPaidPrice in multiple threads, since we're sharing a static numberFormatter... In theory NumberFormatter is thread safe according to docs, but I guess that's within each call, but it does not ensure changes if it would happen in between these 2 calls? Maybe I'm getting it wrong though 😅

Reply:
I think you're right 🤔 
Let me re-think this

Reply:
I've added a NSCache (thread safe) based on locales. I think that should be enough